README.md

# photobiologyWavebands

[![CRAN
version](https://www.r-pkg.org/badges/version-last-release/photobiologyWavebands)](https://cran.r-project.org/package=photobiologyWavebands)
[![cran
checks](https://cranchecks.info/badges/worst/photobiologyWavebands)](https://cran.r-project.org/web/checks/check_results_photobiologyWavebands.html)

Package ‘**photobiologyWavebands**’ is a set of functions and data to be
used together with package ‘**photobiology**’ for calculation of derived
quantities from spectral data. Non-weighed derived quantities represent
summaries of a given range of wavelengths. All constructors default to
ISO standardized definitions when they are available, but other
competing definitions are also available. These additional definitions
include the wavelength ranges used by Landsat satellite imagers, as
these may be needed to approximate remote-sensing-related index
calculations on spectral data.

Derived *biologically effective* quantities are used to quantify the
effect of radiation on different organisms or processes within
organisms. These effects can range from damage to perception of
informational light signals. Except for the definition of the erythemal
and vitamin-D spectra for which CIE definitions exist, those
formulations that we consider the most frequently used are the default.
Weighting functions included in the package are the luminous efficiency
functions for human vision and several biological spectral weighting
functions (BSWFs) used to estimate effective UV doses.

The definitions cover most non-weighted wavebands in common use, and in
addition many weighting functions used in the calculation of effective
irradiances and exposures. The current version of the package puts
emphasis, in the case of weighting functions, on a subset of those in
use, but in the future the package will be expanded to included
responses relevant to additional groups of organisms. Colour-response
and colour-matching functions for human vision and bee vision are
included in package ‘**photobiology**’.

## Installation

Installation of the most recent stable version from CRAN:

``` r
install.packages("photobiologyWavebands")
```

Installation of the current unstable version from Bitbucket:

``` r
# install.packages("devtools")
devtools::install_bitbucket("aphalo/photobiologyWavebands")
```

## Documentation

HTML documentation is available at
(<https://docs.r4photobiology.info/photobiologyWavebands/>), including
an *User Guide*.

News on updates to the different packages of the ‘r4photobiology’ suite
are regularly posted at (<https://www.r4photobiology.info/>).

Two articles introduce the basic ideas behind the design of the suite
and describe its use: Aphalo P. J. (2015)
(<https://doi.org/10.19232/uv4pb.2015.1.14>) and Aphalo P. J. (2016)
(<https://doi.org/10.19232/uv4pb.2016.1.15>).

A book is under preparation, and the draft is currently available at
(<https://leanpub.com/r4photobiology/>).

A handbook written before the suite was developed contains useful
information on the quantification and manipulation of ultraviolet and
visible radiation: Aphalo, P. J., Albert, A., Björn, L. O., McLeod, A.
R., Robson, T. M., & Rosenqvist, E. (Eds.) (2012) Beyond the Visible: A
handbook of best practice in plant UV photobiology (1st ed., p. xxx +
174). Helsinki: University of Helsinki, Department of Biosciences,
Division of Plant Biology. ISBN 978-952-10-8363-1 (PDF),
978-952-10-8362-4 (paperback). PDF file available from
(<http://doi.org/10.31885/9789521083631>).

## Contributing

Pull requests, bug reports, and feature requests are welcome at
(<https://bitbucket.org/aphalo/photobiologyWavebands>).

## Citation

If you use this package to produce scientific or commercial
publications, please cite according to:

``` r
citation("photobiologyWavebands")
#> 
#> To cite package 'photobiologyWavebands' in publications, please use:
#> 
#>   Aphalo, Pedro J. (2015) The r4photobiology suite. UV4Plants Bulletin,
#>   2015:1, 21-29. DOI:10.19232/uv4pb.2015.1.14
#> 
#> A BibTeX entry for LaTeX users is
#> 
#>   @Article{,
#>     author = {Pedro J. Aphalo},
#>     title = {The r4photobiology suite},
#>     journal = {UV4Plants Bulletin},
#>     volume = {2015},
#>     number = {1},
#>     pages = {21-29},
#>     year = {2015},
#>     doi = {10.19232/uv4pb.2015.1.14},
#>   }
```

## License

© 2012-2020 Pedro J. Aphalo (<pedro.aphalo@helsinki.fi>). Released under
the GPL, version 2 or greater. This software carries no warranty of any
kind.


NEWS
photobiologyWavebands 0.4.4 (2020-04-xx)
========================================

* Add NDVI(), normalized difference vegetation index.
* Update docs for 'roxygen2' 7.1.0.

photobiologyWavebands 0.4.3 (2019-03-27)
========================================

* Add waveband definitions for non-standard split of the UVA band with
  constructors UVAsw() and UVAlw().
* Add two new definitions of far-red as used in remote sensing: "RedEdge20" and
  "RedEdge40".
* Minor updates to documentation.

photobiologyWavebands 0.4.2 (2017-03-20)
========================================

Add constructors for frequently used infrared bands, and for lists of infrared
bands.
Add as additional values recognized as argument for parameter std "RS" for
wavelength ranges commonly used in remote sensing to the different existing
and new waveband constructors.
Add constructors for wavelength-range based waveband definitions for the imagers
used in Landsat missions 1 to 8 : "LandsatRBV", "LandsatMSS", "LandsatTM",
"LandsatETM", "LandsatOLI", and "LandsatTIRS".
Add constructor Landsat_bands() returning lists of waveband objects for the
different Landsat missions.
Update warning messages and error handling in existing constructors.
Update Plant_bands() constructor to support CIE's UVA1 and UVA2 definitions.
Update vignette.

photobiologyWavebands 0.4.1
===========================

Update documentation.

photobiologyWavebands 0.4.0
===========================

Remove all definitions of ratios.
Remove vignette with plots of the wavebands.
Remove dependency of vignettes on package ggspectra.

photobiologyWavebands 0.3.4
===========================

Use package ggspectra instead of photobiologygg in vignette for compatibility
with ggplot2 version 2.

photobiologyWavebands 0.3.3
===========================

Minor correction of hinges in definitions to avoid errors at boundaries.

photobiologyWavebands 0.3.2
===========================

Added function and waveband constructor for ICNIRP BSWF.
Rebuilt data and package with photobiology 0.8.0.

photobiologyWavebands 0.3.1
===========================

R_FR_ratio() now uses "Smith10" giving 10 nm bands.

photobiologyWavebands 0.3.0
===========================

Rebuilt data and package with photobiology 0.6.0.
This required some minor edits to the vignettes.

photobiologyWavebands 0.2.8
===========================

Edited the definitions of Red() and Far_red() to include additional
wavelength limits as used by other authors.

Updated User Guide.

photobiologyWavebands 0.2.7
===========================

Some edits to vignettes.

photobiologyWavebands 0.2.6
===========================

Added luminous efficiency functions for human vision, and the constants
used in the definition of Lumen.
Setlow's BSWF is now stored as a response.spct object using the same
variable names as in the rest of the suite.
The Setlow data from TUV has already been converted to energy units,
but this was not noticed when this data was added to the package. This
bug is now corrected.
Updated User Guide with some examples of the calculation of luminous
flux in lux from spectral irradiance data.
Added plots of the luminous efficiency curves to the Data Plots vignette.

photobiologyWavebands 0.2.5
===========================

Updated vignettes.

photobiologyWavebands 0.2.4
===========================

Update to documentation and vignettes.
UVR8-related data and functions moved to package photobiologyPlants.

photobiologyWavebands 0.2.3
===========================

Now all constructors of waveband objects for BSWFs, have formal arguments
w.low and w.high, with values earlier in use, now set as defaults.

Added some references to the documentation.

User Guide edited to match current versions of packages photobiology
and photobiologygg, as well as to update to to the current version.

photobiologyWavebands 0.2.2
===========================

Changed the definition of SWFs functions not to have any
intrinsic wavelength limits, and made this limit
a new formal argument on the waveband constructor functions,
keeping the same defaults. Returned values
should remain unchanged. This bit of flexibility is needed
as different authors have used different limits
for the same BSWFs.

A new formulation for GPAS has been added. The formulation
by Micheletti and Piacentini (2002) is used for
example in the TUV model (version 5.0). The new waveband
constructor is GEN.M().

photobiologyWavebands 0.2.1
===========================

Changed the definition of SWFs to not include wavelengths
shorter than 280 nm, except for the case of UVR8 absorbance.

photobiologyWavebands 0.2.0
===========================

Edits to User Guide for compatibility and rebuilt with photobiology 0.5.1, now required.

photobiologyWavebands 0.1.0
===========================

Merge of photobiologyUV and photobiologyVIS packages.
Merged and updated the vignettes. Added functions for
calculating UV:PAR photon ratios. Added functions for
creating some coherent lists of wavebands (e.g. all ISO
defined colours).

********************
*  photobiologyUV  *
********************

photobiologyUV 0.2.8
====================

Added setting of wb.label in functions defining wavebands.

photobiologyUV 0.2.7
====================

Fixed a bug introduced in version 0.2.6: wrong name used for wavebands created by UVC().
photobiologyUV 0.2.6
====================

Added "medical" definitions of UVB and UVB.

photobiologyUV 0.2.5
====================

Added "methane from pectin" BSWF of McLeod et al. (2008)

photobiologyUV 0.2.4
====================

Minor changes, with no effect on functionality

photobiologyUV 0.2.3
====================

Revised User Guide and added Catalogue of plot of BSWFs.
No changes to the R code.

photobiologyUV 0.2.2
====================


R/red.r
#' Constructor of red waveband
#'
#' Red radiation according to "ISO" (610-760 nm) or as commonly defined in plant
#' photobiology, "Smith10" (655-665 nm), "Smith20" (650-670 nm), "Inada"
#' (600-700 nm), "Warrington" (625-675 nm), and "Sellaro" (620-680 nm). No
#' weighting applied.
#'
#' @param std a character string, "ISO", "Smith10", "Smith20", "Inada",
#'   "Warrington", "Sellaro", "RS", or for Landsat imagers "LandsatRBV", etc.
#'
#' @return a waveband object wavelength defining a wavelength range.
#'
#' @export
#'
#' @references Aphalo, P. J., Albert, A., Björn, L. O., McLeod, A. R., Robson,
#' T. M., Rosenqvist, E. (Eds.). (2012). Beyond the Visible: A handbook of best
#' practice in plant UV photobiology (1st ed., p. xxx + 174). Helsinki:
#' University of Helsinki, Department of Biosciences, Division of Plant Biology.
#' ISBN 978-952-10-8363-1 (PDF), 978-952-10-8362-4 (paperback). Open access PDF
#' download available at https://hdl.handle.net/10138/37558
#'
#' ISO (2007) Space environment (natural and artificial) - Process for
#' determining solar irradiances. ISO Standard 21348. ISO, Geneva.
#'
#' Murakami, K., Aiga I. (1994) Red/Far-red photon flux ratio used as an index
#' number for morphological control of plant growth under artificial lighting
#' conditions. Proc. Int. Symp. Artificial Lighting, Acta Horticulturae, 418,
#' ISHS 1997.
#'
#' Sellaro, R., Crepy, M., Trupkin, S. A., Karayekov, E., Buchovsky, A. S.,
#' Rossi, C., & Casal, J. J. (2010). Cryptochrome as a sensor of the blue/green
#' ratio of natural radiation in Arabidopsis. Plant physiology, 154(1), 401-409.
#' doi:10.1104/pp.110.160820
#'
#' Smith, H. (1982) Light quality, photoperception and plant strategy. Annual
#' Review of Plant Physiology, 33:481-518.
#'
#' @seealso \code{\link[photobiology]{waveband}}
#'
#' @examples
#' Red()
#' Red("ISO")
#' Red("Smith")
#' Red("Sellaro")
#'
#' @family unweighted wavebands
#'
Red <- function(std="ISO") {
  label="Red"
  if (std=="Smith") {
    warning("The definition of 'Smith' defaults to 'Smith10', to restore old behaviour use 'Smith20'.")
    std <- "Smith10"
  }
  if (std=="ISO") {
    new_waveband(610, 760, wb.name=paste("Red", std, sep="."), wb.label=label)
  } else if (std=="Smith20") {
    new_waveband(650, 670, wb.name=paste("Red", std, sep="."), wb.label="R")
  } else if (std=="Smith10") {
    new_waveband(655, 665, wb.name=paste("Red", std, sep="."), wb.label="R")
  } else if (std=="Inada") {
    new_waveband(600, 700, wb.name=paste("Red", std, sep="."), wb.label="R")
  } else if (std=="Warrington") {
    new_waveband(625, 675, wb.name=paste("Red", std, sep="."), wb.label="R")
  } else if (std=="Sellaro") {
    new_waveband(620, 680, wb.name=paste("Red", std, sep="."), wb.label=label)
  } else if (std=="LandsatOLI" || std=="RS") {
    new_waveband(630, 680, wb.name=paste("Red", std, sep="."), wb.label=label)
  } else if (std=="LandsatETM" || std=="LandsatTM") {
    new_waveband(630, 690, wb.name=paste("Red", std, sep="."), wb.label=label)
  } else if (std=="LandsatMSS") {
    new_waveband(600, 700, wb.name=paste("Red", std, sep="."), wb.label=label)
  } else if (std=="LandsatRBV") {
    new_waveband(580, 680, wb.name=paste("Red", std, sep="."), wb.label=label)
  } else {
    warning("'std' = '", std, "' not implemented.")
    NA
  }
}
